upgrader.go

```diff
@@ -213,6 +214,37 @@ func (u *Upgrader) Upgrade() error {
 	}
 }
 
+// WaitForParentExit wait until parent exit if parent exists
+// otherwise returns directly
+// returns nil when OK, non nil when NG
+func (u *Upgrader) WaitForParentExit(ctx context.Context) error {
+	if u.parent == nil {
+		return nil
+	}
+
+	if u.parentErr != nil {
+		return u.parentErr
+	}
+
+	select {
+	default:
+		return errors.New("upgrade in progress")
+	case u.upgradeSem <- struct{}{}:
+	}
+
+	defer func() {
+		<-u.upgradeSem
+	}()
+
+	select {
+	case err := <-u.parent.exited:
```

Comment:
This has the effect of "stealing" err from `Upgrade`, which leads to inconsistent behaviour.